src/uct/api/uct.h

```diff
@@ -2697,6 +2859,70 @@ UCT_INLINE_API unsigned uct_iface_progress(uct_iface_h iface)
 }
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Open a connection manager.
+ *
+ * Open a specific connection manager. All client server connection
+ * establishment operations are performed in the context of a specific
+ * connection manager.
+ * @note This is an alternative API for
+ *       @ref uct_iface_open_mode::UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER and
+ *       @ref uct_iface_open_mode::UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT .
+ *
+ * @param [in]  component   Component on which to open the connection manager,
+ *                          as returned from @ref uct_query_components.
+ * @param [in]  worker      Worker on which to open the connection manager.
+ * @param [out] cm_p        Filled with a handle to the connection manager.
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_cm_open(uct_component_h component, uct_worker_h worker, uct_cm_h *cm_p);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Close a connection manager.
+ *
+ * @param [in]  cm    Connection manager to close.
+ */
+void uct_cm_close(uct_cm_h cm);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Get connection manager attributes.
+ *
+ * @param [in]  cm      Connection manager to query.
+ * @param [out] cm_attr Filled with connection manager attributes.
+ */
+ucs_status_t uct_cm_query(uct_cm_h cm, uct_cm_attr_t *cm_attr);
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Create a new transport listener object.
+ *
+ * @param [in]  params      User defined @ref uct_listener_params_t
+ *                          configurations for the @a listener_p.
+ * @param [out] listener_p  Filled with handle to the new listener.
+ *
+ *
+ * @return Error code.
+ */
+ucs_status_t uct_listener_create(const uct_listener_params_t *params,
```

Comment:
Shell we move the `uct_cm_h cm` out of `uct_listener_params_t` and make this consistent with `uct_cm_query`